pkg/render/node.go

```diff
@@ -891,49 +853,81 @@ func (c *nodeComponent) nodeEnvVars() []v1.EnvVar {
-			// IPv6 Auto-detection is enabled.
-			nodeEnv = append(nodeEnv, v1.EnvVar{Name: "IP6", Value: "autodetect"})
-			nodeEnv = append(nodeEnv, v1.EnvVar{Name: "IP6_AUTODETECTION_METHOD", Value: v6Method})
-			nodeEnv = append(nodeEnv, v1.EnvVar{Name: "FELIX_IPV6SUPPORT", Value: "true"})
-
-			// Set CALICO_ROUTER_ID to "hash" if IPv6 only.
-			if v4Method == "" {
-				nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_ROUTER_ID", Value: "hash"})
-			}
-		} else {
-			// IPv6 Auto-detection is disabled.
-			nodeEnv = append(nodeEnv, v1.EnvVar{Name: "IP6", Value: "none"})
-			nodeEnv = append(nodeEnv, v1.EnvVar{Name: "FELIX_IPV6SUPPORT", Value: "false"})
-		}
-
-		if v6pool != nil {
+		// Configure IPv6 pool.
+		if v6pool := GetIPv6Pool(c.cr.Spec.CalicoNetwork.IPPools); v6pool != nil {
 			nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_IPV6POOL_CIDR", Value: v6pool.CIDR})
 
 			if v6pool.BlockSize != nil {
 				nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_IPV6POOL_BLOCK_SIZE", Value: fmt.Sprintf("%d", *v6pool.BlockSize)})
 			}
 			if v6pool.NATOutgoing == operator.NATOutgoingDisabled {
-				nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_IPV6POOL_NAT_OUTGOING",
-					Value: "false"})
+				nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_IPV6POOL_NAT_OUTGOING", Value: "false"})
 			}
 			if v6pool.NodeSelector != "" {
-				nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_IPV6POOL_NODE_SELECTOR",
-					Value: v6pool.NodeSelector})
+				nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_IPV6POOL_NODE_SELECTOR", Value: v6pool.NodeSelector})
 			}
 		}
+
+	}
+
+	// Determine MTU to use. If specified explicitly, use that. Otherwise, set defaults based on an overall
+	// MTU of 1460.
+	ipipMtu := "1440"
+	vxlanMtu := "1410"
+	wireguardMtu := "1400"
+	if c.cr.Spec.CalicoNetwork.MTU != nil {
+		ipipMtu = strconv.Itoa(int(*c.cr.Spec.CalicoNetwork.MTU))
+		vxlanMtu = strconv.Itoa(int(*c.cr.Spec.CalicoNetwork.MTU))
+		wireguardMtu = strconv.Itoa(int(*c.cr.Spec.CalicoNetwork.MTU))
+	}
+	nodeEnv = append(nodeEnv, v1.EnvVar{Name: "FELIX_VXLANMTU", Value: vxlanMtu})
+	nodeEnv = append(nodeEnv, v1.EnvVar{Name: "FELIX_WIREGUARDMTU", Value: wireguardMtu})
+
+	// Configure whether or not BGP should be enabled.
+	if !bgpEnabled(c.cr) {
+		// If BGP is disabled, then set the networking backend to "vxlan". This means that BIRD will be
+		// disabled, and VXLAN will optionally be configurable via IP pools. This works even if you're
+		// not actually using Calico VXLAN for networking (e.g., via an alternative CNI plugin).
+		nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_NETWORKING_BACKEND", Value: "vxlan"})
+	} else {
+		// BGP is enabled.
+		nodeEnv = append(nodeEnv, v1.EnvVar{Name: "CALICO_NETWORKING_BACKEND", Value: "bird"})
+		nodeEnv = append(nodeEnv, v1.EnvVar{Name: "FELIX_IPINIPMTU", Value: ipipMtu})
+	}
+
+	// IPv4 auto-detection configuration.
+	v4Method := getAutodetectionMethod(c.cr.Spec.CalicoNetwork.NodeAddressAutodetectionV4)
```

Comment:
The getAutodetectionMethod calls need to be protected with `CalicoNetwork != nil`